tutorial.md

```diff
@@ -1,101 +1,85 @@
-file name if need be.
+10. In the "Output" section, please select an "Output Directory" where the capture file will be stored. The "File Name" should be auto-filled. You can also edit the file name.
 
-In the GPU section, click on "Select" to bring up the counter selection dialog.
-Click on "default" to select the set of default counters and press "OK".
+11. In the GPU section, click on "Select" to bring up the counter selection dialog. Click on "default" to select the set of default counters and press "OK".
 
-<img src="../images/system_profiler/counter-config.png" width="500px">
+    <img src="../images/system_profiler/counter-config.png" width="500px">
 
-Once all this is done, press "OK". This starts the selected app on the Android
-device, and creates a pop-up window with a "Start" button. Click the "Start"
-button to start the performance capture, and wait for a couple of seconds for
+12. Once all this is done, press "OK". This starts the selected app on the Android device, and creates a pop-up window with a "Start" button. Click the "Start" button to start the performance capture, and wait for a couple of seconds for
 the trace to terminate.
 
-Once the capture is completed, click on "Open Trace". This leads to a view
-similar to [systrace](https://developer.android.com/studio/profile/systrace).
+13. Once the capture is completed, click on "Open Trace". This leads to a view similar to [systrace](https://developer.android.com/studio/profile/systrace).
 
-<img src="../images/system_profiler/system-profile.png" width="500px">
+    <img src="../images/system_profiler/system-profile.png" width="500px">
 
-Compare to systrace, there are now additional information relating to the GPU.
-The GPU Queue section shows when GPU commands are submitted and when the GPU
-actually starts work.  The GPU Counters shows various performance counters to
-help optimize the graphics works in the app.
+    Compare to systrace, there are now additional information relating to the GPU. The GPU Queue section shows when GPU commands are submitted and when the GPU actually starts work.  The GPU Counters shows various performance counters to help optimize the graphics works in the app.
 
 ## Validate Vulkan on Android
 
-The official Android documentation contains a section about
-[Vulkan Validation layers](https://developer.android.com/ndk/guides/graphics/validation-layer).
+The official Android documentation contains a section about [Vulkan Validation layers](https://developer.android.com/ndk/guides/graphics/validation-layer).
 
-If your app does not already have a setup to run with the validation layers, you
-can use the following commands to force it to run with the validation layers
-shipped in the GAPID apk:
+If your app does not already have a setup to run with the validation layers, you can use the following commands to force it to run with the validation layers shipped in the GAPID apk:
 
 ```sh
 app_package=<YOUR APP PACKAGE NAME HERE>
```

Comment:
(GH doesn't allow me to add the comment below in the already existing part of the file... This comment should be in the `sh` section below).

Remove lines 91-93 (form "The order of layers matter.. to "Once NDK r21...") and remove the `#` on line 94. It should be:
```
...
adb shell settings put global gpu_debug_layer_app com.google.android.gapid.${abi}
adb shell settings put global gpu_debug_layers VK_LAYER_KHRONOS_validation
...
```

Reply:
See https://github.com/google/agi/pull/182/commits/62bf1b179b61c5e8ae25464df3961aa218ff1bd8